cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -1629,6 +1619,61 @@ public void addSubscriptionTags(String[] tagIds) {
         addSubscriptionTagTrackingConsent(tagIds);
     }
 
+    public void addSubscriptionTopic(String topicId, CompletionListener completionListener) {
+        this.getSubscriptionId(subscriptionId -> {
+            Set<String> topics = this.getSubscriptionTopics();
+            if (topics.contains(topicId)) {
+                completionListener.onComplete();
+                return;
+            }
+            topics.add(topicId);
+            List<String> list = new ArrayList<>(topics);
+            String[] topicsArray = (String[]) list.toArray();
+            this.setSubscriptionTopics(topicsArray, null);
+            if (subscriptionId != null) {
```

Comment:
is there a reason for doing the code above this line, if this is the case? if not maybe an early exit, not sure if we can even have this case